### PR TITLE
feat(TDI-46151): Filter data based on return value in subquery in source database before pushing it down to a tDBInput to execute

### PR DESCRIPTION
**Pull request by Tanya5091**
[JIRA ticket](https://jira.talendforge.org/browse/TDI-46151)
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tELTOutput/tELTOutput_java.xml [hunk 1]
```diff
@@ -209,6 +209,7 @@
   <RETURNS>
     <RETURN NAME="NB_LINE"          TYPE="id_Integer" AVAILABILITY="AFTER"/>
     <RETURN NAME="NB_LINE_INSERTED" TYPE="id_Integer" AVAILABILITY="AFTER" SHOW_IF="DATA_ACTION=='INSERT'"/>
+    <RETURN NAME="QUERY" TYPE="String" AVAILABILITY="AFTER"/>
```
Comment (pyzhou-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tELTOutput/tELTOutput_java.xml:212]: I think the type should be id_String here

File: main/plugins/org.talend.designer.components.localprovider/components/tELTOutput/tELTOutput_main.javajet [hunk 2]
```diff
@@ -81,53 +83,59 @@ skeleton="../templates/db_output_bulk.skeleton"
     String jdbcUrl  = null;
     String dbuser = null;
     boolean useExistingConn = false;
+    boolean dryRun = false;
+    if(previousNode != null) {
```
Comment (pyzhou-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tELTOutput/tELTOutput_main.javajet:87]: Better to keep clean logic in javajet, as javajet itself is mysterious enough.
if(previousNode != null) {
    if(!dryRun){
//code
}else{
//code
}
}

**Review by pyzhou-talend — CHANGES_REQUESTED** [review]

**Review by pyzhou-talend — DISMISSED** [review]
LGTM

File: main/plugins/org.talend.designer.components.localprovider/components/tELTMSSqlOutput/tELTMSSqlOutput_main.javajet [hunk 3]
```diff
@@ -43,11 +43,11 @@ imports="
 	    String previousComponentName = previousNode.getUniqueName();
 		dbtable = connection.getName();
 		uniqueNameConnection = connection.getUniqueName();
-		
+
 		%>
 		select_query_<%=cid %> = (String) globalMap.get("<%=previousComponentName%>"+"QUERY"+"<%=uniqueNameConnection%>");
 		selectQueryColumnsName_<%=cid %> = (String) globalMap.get("<%=previousComponentName%>"+"QUERY_COLUMNS_NAME"+"<%=uniqueNameConnection%>");
-		
+		String query_<%=cid%> = null;
```
Comment (olhava) [inline main/plugins/org.talend.designer.components.localprovider/components/tELTMSSqlOutput/tELTMSSqlOutput_main.javajet:50]: Maybe need to initialize this variable in any case? (Not only if the check with incoming connections is true)

**Review by olhava — APPROVED** [review]

File: main/plugins/org.talend.designer.components.localprovider/components/tELTMSSqlOutput/tELTMSSqlOutput_main.javajet (hunk 3 above)
Comment (Tanya5091) [reply main/plugins/org.talend.designer.components.localprovider/components/tELTMSSqlOutput/tELTMSSqlOutput_main.javajet:50]: Okay, sounds right

**Review by olhava — APPROVED** [review]

**Review by ypiel-talend — APPROVED** [review]
LGTM

**Pull request merged**